File: Codility/TapeEquilibrium/TapeEquilibrium/ViewController.m
//
//  ViewController.m
//  TapeEquilibrium
//
//  Created by Ancil on 9/14/15.
//  Copyright (c) 2015 Ancil Marshall. All rights reserved.
//

#import "ViewController.h"

@interface Square : NSObject

@end

@implementation Square

-(instancetype)init;
{
    self = [super init];
    NSLog(@"Init");
    return self;
}

-(void)doSomething;
{
    NSLog(@"DoSomething");
}

@end


@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];

    Square* sq = nil;
    [sq doSomething];
    
    NSArray* arr = [NSArray arrayWithObjects:nil];
    
    id myobjet;
    
    
    //Square* sq = [[Square alloc] init];

}


@end

File: LocationMaps/CLLocationManagerDemoSwift/CLLocationManagerDemoSwift/ViewController.swift
//
//  ViewController.swift
//  CLLocationManagerDemoSwift
//
//  Created by Ancil on 10/4/15.
//  Copyright © 2015 Ancil Marshall. All rights reserved.
//

import UIKit
import CoreLocation

class ViewController: UIViewController {

    @IBOutlet weak var lonLabel:UILabel!
    @IBOutlet weak var latLabel:UILabel!
    @IBOutlet weak var markLabel:UILabel!
    
    let locationManager = CLLocationManager()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        if  !CLLocationManager.locationServicesEnabled() {
            return
        }
        locationManager.delegate = self
        beginLocationUpdatesIfPossible()
    }
    
    
    func beginLocationUpdatesIfPossible() {
        
        switch CLLocationManager.authorizationStatus() {
            
        case .Denied, .Restricted:
            return
            
        case .NotDetermined:
            locationManager.requestAlwaysAuthorization()
            
        case .AuthorizedAlways, .AuthorizedWhenInUse:
            break
        }
        
        locationManager.startUpdatingLocation()
    }

}


extension ViewController: CLLocationManagerDelegate {
    
    func locationManager(manager: CLLocationManager, didChangeAuthorizationStatus status: CLAuthorizationStatus) {
        
        beginLocationUpdatesIfPossible()
    }
    
    func locationManager(manager: CLLocationManager, didUpdateLocations locations: [CLLocation]) {
        
        let location = locations.first!
        lonLabel.text = "\(location.coordinate.longitude)"
        latLabel.text = "\(location.coordinate.latitude)"
    }
}

